src/pull.go

```diff
@@ -82,6 +92,76 @@ func (g *Commands) Pull(byId bool) error {
 	return g.playPullChanges(nonConflicts, g.opts.Exports, opMap)
 }
 
+func autoRenameClashes(g *Commands, clashes []*Change) error {
+	type renameOp struct {
+		newName      string
+		change       *Change
+		originalPath string
+	}
+	clashesMap := map[string][]*Change{}
+
+	for _, clash := range clashes {
+		group := clashesMap[clash.Path]
+		if clash.Src == nil {
+			continue
+		}
+		clashesMap[clash.Path] = append(group, clash)
+	}
+
+	renames := make([]renameOp, 0, len(clashesMap)) // we will have at least len(clashesMap) renames
```

Comment:
I'm not sure what you mean. For each clash we will have at least one rename so for sure we will have at least len(clashesMap) renames. This is a micro optimization not to reallocate memory each time we do append (which would happen otherwise as I understand this). But let's face it - this is not performance critical app so this optimization does not really matter that much, right?